### ISSUE TITLE
Allow re-arrangement of UB level groups

### ISSUE THREAD

**Issue by Richard-L** [LostArtefacts/TRX#1048]
The recent UB Saturn port has resurfaced an old interview with Phil Campbell, the main designer of the UB levels, in which he says:

> Atlantis WAS indeed a return to clear out the remaining baddies, and was originally intended to be released as 'expert' levels. **They should have come before the cat levels on the disk and you should have started the cat levels with no inventory.**

Source: http://www.trinity.whsites.net/tomb/misc/intphil.html

We'd talked about the re-arrangement of the levels a few times in the past, and I thought I'd make an issue for it. Since Campbell mentions the order of the levels "on the disc", I wonder if there's a chance the level sequence resulted purely by alphabetic order, and is thus without any artistic or narrative intent?

Essentially there are three chapters in TR1:
1. Atlantean Scion (=base game)
2. Unfinished Business
3. Shadow of the Cat

I think it would be nice to make this order possible. I would even suggest to potentially include all three chapters under one umbrella, allow the player to unlock one after the other, and lose the necessary `-gold` parameter with the executable. But maybe that's beyond the scope of this issue.

More on topic: Would it be feasible to swap the two groups of UB levels? It would mean that "Atlantean Stronghold" ideally were to inherit the inventory the player has after beating "The Great Pyramid". It would then also mean that the inventory is reset for "Return to Egypt", as Campbell mentions.

Some questions I had:
- Should the final statistics screen show after each of the three scenarios? Should it regard only the chapter the player just completed, or should it include the stats from the previous levels as well? I would say focus only on one chapter at hand. Potentially adding a line saying "Congratulations, you've unlocked `<chapter>`".
- Should the final FMV play after beating "The Great Pyramid" or after "The Hive"?

**Comment by rr-** [comment]
I believe this is a quite niche use case, and since supporting it would require significant development effort, it might be best for people who are willing to play it, to personally tailor the gameflow file to their liking.

**Comment by Richard-L** [comment]
Unfortunate that it would mean a bigger undertaking code wise. I understand then it may not be desirable.

I could try it out by tailoring my gameflow like you suggested. I think I know how to re-arrange the levels. Is it also possible to tell it when to inherit or reset the inventory?

**Comment by walkawayy** [comment]
> Unfortunate that it would mean a bigger undertaking code wise. I understand then it may not be desirable.
> 
> I could try it out by tailoring my gameflow like you suggested. I think I know how to re-arrange the levels. Is it also possible to tell it when to inherit or reset the inventory?

Yes check here: https://github.com/LostArtefacts/TR1X/blob/develop/GAMEFLOW.md

There are [sequences ](https://github.com/LostArtefacts/TR1X/blob/develop/GAMEFLOW.md#sequences ) like `remove_ammo`, `remove_guns`, etc.

**Comment by CharlesLove** [comment]
I made a quick edit to the gameflow to better match the remastered level order. I've also reset the inventory to just pistols when returning to Egypt. Just unzip and drop the "TR1X_gameflow_ub.json5" file in your "cfg" folder.

[TR1X_ub_remastered_gameflow.zip](https://github.com/LostArtefacts/TR1X/files/14497743/TR1X_ub_remastered_gameflow.zip)

**Comment by Richard-L** [comment]
@CharlesLove only just tried this - it's excellent 👌

**Comment by pas-de-2** [comment]
> I made a quick edit to the gameflow to better match the remastered level order. I've also reset the inventory to just pistols when returning to Egypt. Just unzip and drop the "TR1X_gameflow_ub.json5" file in your "cfg" folder.
> 
> [TR1X_ub_remastered_gameflow.zip](https://github.com/LostArtefacts/TR1X/files/14497743/TR1X_ub_remastered_gameflow.zip)

Updated for the new gameflow format:
[TR1X_gameflow_ub_campbell.zip](https://github.com/user-attachments/files/18800494/TR1X_gameflow_ub_campbell.zip)

EDIT (5/4/26) : Updated again for TRX 1.6:
[TRX_gameflow_ub_campbell_1.6.zip](https://github.com/user-attachments/files/27371775/TRX_gameflow_ub_campbell_1.6.zip)

**Comment by rr-** [comment]
Related: #1170
